src/shared/edit-util.c

```diff
@@ -18,18 +18,100 @@
 #include "strv.h"
 #include "tmpfile-util.h"
 
-void edit_file_free_all(EditFile **ef) {
-        assert(ef);
+static void edit_file_context_unlink_all(EditFileContext *context) {
+        assert(context);
+
+        EDIT_FILES_FOREACH(i, context)
+                if (i->temp)
+                        (void) unlink(i->temp);
+}
+
+void edit_file_context_done(EditFileContext *context) {
+        assert(context);
+
+        EDIT_FILES_FOREACH(i, context) {
+                if (i->temp) {
+                        (void) unlink(i->temp);
+                        free(i->temp);
+                }
+
+                if (context->remove_parent) {
+                        _cleanup_free_ char *parent = NULL;
+                        int r;
+
+                        r = path_extract_directory(i->path, &parent);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to extract directory from '%s': %m", i->path);
```

Comment:
Now the error will be ignored. Please downgrade the log level, and include `", ignoring"` in the log message.